File: src/main/java/gc/temp/SingleNumber.java
package gc.temp;

public class SingleNumber {

	public static void main(String[] args) {
//			System.out.println(12^12);
		int[] a = {1,1,2,3,4,2,3,4,6};
		System.out.println(findSingleNumber(a));
	}

	private static int findSingleNumber(int[] a) {
		int result = 0;
		for(int i:a){
			result ^=i;
		}
		return result;
	}

}

File: src/main/scala/array/SingleNumber.scala
package array

/**
  * Created by guchao on 18/12/21.
  * leetcode 136
  */
object SingleNumber {
  def singleNumber(nums: Array[Int]): Int = {
    var result = 0
    for(num<-nums){
      result = result ^ num
    }
    result
  }

  def main(args: Array[String]): Unit = {
      println(singleNumber(Array(4,1,2,1,2)))
  }

}
